docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+```json
+{
+    "schemaVersion": 2,
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:ef87d9f0...",
+        "size": 1510
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
+            "digest": "sha256:f69ae40...",
+            "size": 204812,
+            "annotations": {
+                "org.opencontainers.image.enc.pubopts": "eyJjaXBoZXIiOi...",
+                "org.opencontainers.image.enc.keys.provider.attestation-agent": "eyJwcm9..."
+            }
+        }
+    ]
+}
+```
+
+### Image signing
+
+The container image signing standard used by the confidential containers V1 is follow the signing mechanism used in [containers/image](https://github.com/containers/image). At present, the only defined cryptographic signature format is a OpenPGP signature (RFC 4880). The signing process will not modify any content of the container image, including layer, config, manifest, etc. Therefore, the signature will not change the image manifest digest used to identify the image.
+
+The container signature under this standard is essentially a blob, which consists of two parts:
+
+- JSON payload
+
+- Cryptographic signature of JSON payload
+
+The JSON payload contains the manifest digest and identity information of the container image. Examples are as follows:
+
+```json
+{
+    "critical": {
+        "type": "atomic container signature",
+        "image": {
+            "docker-manifest-digest": "sha256:817a12c32a39be..."
+        },
+        "identity": {
+            "docker-reference": "docker.io/library/busybox:latest"
+        }
+    },
+    "optional": {
+        "creator": "some software package v1.0.1-35",
+        "timestamp": 1483228800,
+    }
+}
+```
+
+The process of signing a container image is as follows:
+
+1. Create a signature JSON payload and write the manifest digest together with some otheridentity information of the container image.
+2. Use the private key to sign the JSON payload generated in the previous step, and package the cryptographic signature together with the JSON payload into a signature claim file.
+
+The signature claim file is what we usually call the container image signature file. Its directory structure and file name format are as follows:
+
+```
+.
+└── busybox@sha256=817a12c32a39bbe394944ba49de563e085f1d3c5266eb8e9723256bc4448680e
+    └── signature-1
+```
+
+The meaning of directory name is `image-name@digest-algo=digest-value`, multiple signature files can exist in the directory, which means that the same container image can have multiple signatures and be numbered and stored.
```

Comment:
Not sure about "and be numbered and stored". It looks like this applies to signatures, not container images, right? Why would there be multiple signatures? If you are talking about multiple revisions of the container, then the sha256 would be different, so I guess that's not it.

Or are you considering multiple signatures with different algorithms?

Reply:
Because the sigstore may be a public space, the signatures generated by the keys of different users in the same container may be stored in the same sigstore.